src/librustc_resolve/lib.rs

```diff
@@ -2981,50 +2890,40 @@ impl<'a> Resolver<'a> {
             })
         }
 
-        SuggestedCandidates {
-            name: lookup_name.as_str().to_string(),
-            candidates: lookup_results,
-        }
+        candidates
     }
 
     fn record_def(&mut self, node_id: NodeId, resolution: PathResolution) {
         debug!("(recording def) recording {:?} for {}", resolution, node_id);
+        assert!(resolution.depth == 0 || resolution.base_def != Def::Err);
```

Comment:
This adds an ICE to this example:
```rust
fn foo<T>() {
    const FOO: T::Item = ();
}
```
I'll change it locally to `if resolution.base_def == Def::Err { resolution.depth = 0; }`.